docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -1098,6 +1098,186 @@ public class MyService {
 In the above example, any code calling the `doSomething()` method triggers interception - in this case, the method becomes transactional.
 This is regardless of whether the invocation originated directly from the `MyService` bean (such as `MyService#doSomethingElse`) or from some other bean.
 
+=== Intercepting Producer Methods and Synthetic Beans
+
+By default, interception is only supported for managed beans (also known as class-based beans).
+To support interception of producer methods and synthetic beans, the CDI specification includes an `InterceptionFactory`, which is a runtime oriented concept and therefore cannot be supported in Quarkus.
+
+Instead, Quarkus has its own API: `InterceptionProxy` and `@BindingsSource`.
+The `InterceptionProxy` is very similar to `InterceptionFactory`: it creates a proxy that applies `@AroundInvoke` interceptors before forwarding the method call to the target instance.
+
+[source,java]
+----
+import io.quarkus.arc.InterceptionProxy;
+
+@ApplicationScoped
+class MyProducer {
+    @Produces
+    MyClass produce(InterceptionProxy<MyClass> proxy) { // <1>
+        return proxy.create(new MyClass()); // <2>
+    }
+}
+----
+
+<1> Declares an injection point of type `InterceptionProxy<MyClass>`.
+    This means that at build time, a subclass of `MyClass` is generated that does the interception and forwarding.
+    Note that the type argument must be identical to the return type of the producer method.
+<2> Creates an instance of the interception proxy for the given instance of `MyClass`.
+    The method calls will be forwarded to this target instance after all interceptors run.
+
+In this example, interceptor bindings are read from the `MyClass` class.
+
+Note that `InterceptionProxy` only supports `@AroundInvoke` interceptors declared on interceptor classes.
```

Comment:
This would deserve an AsciiDoc [admonition](https://docs.asciidoctor.org/asciidoc/latest/blocks/admonitions/), NOTE or IMPORTANT...

Reply:
Done.